python/lsst/ts/audiotrigger/read_serial_temp_scanner.py

```diff
@@ -0,0 +1,313 @@
+        self.simulation_mode = simulation_mode
+
+        if not self.simulation_mode:
+            super().__init__(
+                log=self.log,
+                port=port,
+                host=host,
+                connect_callback=None,
+                monitor_connection_interval=0,
+                name="",
+                encoding=encoding,
+                terminator=terminator,
+            )
+
+        self.serial = serial
+        self.sensor_dict = {}
+        self.sample_wait_time = sample_wait_time
+
+        # Fan sensor
+        self.fan_sensor = ""
+        self.fan_gpio = None
+        self.fan_turn_on_temp = 0
+        self.fan_turn_off_temp = 0
+
+        self.latest_data = {sensor_name: 0 for sensor_name in self.sensor_dict}
+        self.rolling_temperature = [0 for _ in range(temperature_windows)]
+
+        if self.simulation_mode:
+            self.pi = MockPio()
+        else:
+            self.pi = pigpio.pi()
+
+        self.configured = False
+        self.first_run = True
+        self.encoding = encoding
+        self.port = port
+        self.host = host
+        self.loop = asyncio.get_running_loop()
+        self.config()
+
+    def config(self):
+        """Configure the temp_scanner."""
+        # TODO: read config .yaml instead
+        PORT = "/dev/ttyUSB0"
+        BAUDRATE = 19200
+        self.sensor_dict = OrderedDict(
+            {
+                "C01": "Ambient",
+                "C02": "Laser",
+                "C03": "FC",
+                "C04": "A",
+                "C05": "B",
+                "C06": "C",
+                "C07": "D",
+                "C08": "E",
+            }
+        )  # These will need to be in some configuration file
+
+        # Define serial connection
+        if self.simulation_mode:
+            self.serial = MockThermalSensor()
+        else:
+            self.serial = serial.Serial(
+                port=PORT,
+                baudrate=BAUDRATE,
+                parity=serial.PARITY_NONE,
+                stopbits=serial.STOPBITS_ONE,
+                bytesize=serial.EIGHTBITS,
+                timeout=1,
+            )
+
+        # fan sensor
+        self.fan_sensor = "Ambient"
+        self.fan_gpio = 4
+        self.fan_turn_on_temp = 25
+        hysteresis = 2
+        self.fan_turn_off_temp = self.fan_turn_on_temp - hysteresis
+        self.configured = True
+
+    async def start(self, **kwargs):
+        """Start reading the temperature channels."""
+        self.task = asyncio.ensure_future(self.serial_temperature_task())
+        await super().start(kwargs=kwargs)
+
+    async def write_if_connected(self, string_to_write):
+        """Write message if connected."""
+        if self.simulation_mode:
+            return
+        if self.connected:
+            await self.write_str(string_to_write)
+
+    async def set_fan(self, setting):
+        """Set the fan on/off.
+
+        Parameters
+        ----------
+        setting : `int`
```

Comment:
Shouldn't this be a `Fan` enum instead? I noticed you haven't enabled mypy so this will not fail.